### PR TITLE
Change Centos target nodes network configuration to keyfile format

### PR DESCRIPTION
**Pull request by Sunnatillo**
These PR changes target nodes network configuration to keyfile format

### CODE REVIEW

File: tests/roles/run_tests/templates/cluster-template-controlplane-kubeadm-config-centos.yaml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 # CentOS specific controlplane kubeadm config
 preKubeadmCommands:
   - systemctl restart NetworkManager.service
-  - nmcli connection load /etc/sysconfig/network-scripts/ifcfg-eth0
-  - nmcli connection up filename /etc/sysconfig/network-scripts/ifcfg-eth0
+  - nmcli con add load /etc/NetworkManager/system-connections/eth0.nmconnection
+  - nmcli connection up filename /etc/NetworkManager/system-connections/eth0.nmconnection
+  - nmcli con add load /etc/NetworkManager/system-connections/{{ IRONIC_ENDPOINT_BRIDGE }}.nmconnection
+  - nmcli connection up filename /etc/NetworkManager/system-connections/{{ IRONIC_ENDPOINT_BRIDGE }}.nmconnection
```
Comment (lentzi90) [inline tests/roles/run_tests/templates/cluster-template-controlplane-kubeadm-config-centos.yaml:7]: This looks weird to me, like there are multiple commands on the same line or something. For example I think it should be `nmcli con load` (without the `add`).

```suggestion
  - nmcli connection load /etc/NetworkManager/system-connections/eth0.nmconnection
  - nmcli connection up eth0
  - nmcli connection load /etc/NetworkManager/system-connections/{{ IRONIC_ENDPOINT_BRIDGE }}.nmconnection
  - nmcli connection up {{ IRONIC_ENDPOINT_BRIDGE }}
```

File: tests/roles/run_tests/templates/cluster-template-workers-kubeadm-config-centos.yaml [hunk 2]
```diff
@@ -1,8 +1,10 @@
 # CentOS specific worker kubeadm config
 preKubeadmCommands:
   - systemctl restart NetworkManager.service
-  - nmcli connection load /etc/sysconfig/network-scripts/ifcfg-eth0
-  - nmcli connection up filename /etc/sysconfig/network-scripts/ifcfg-eth0
+  - nmcli con add load /etc/NetworkManager/system-connections/eth0.nmconnection
+  - nmcli connection up filename /etc/NetworkManager/system-connections/eth0.nmconnection
+  - nmcli con add load /etc/NetworkManager/system-connections/{{ IRONIC_ENDPOINT_BRIDGE }}.nmconnection
+  - nmcli connection up filename /etc/NetworkManager/system-connections/{{ IRONIC_ENDPOINT_BRIDGE }}.nmconnection
```
Comment (lentzi90) [inline tests/roles/run_tests/templates/cluster-template-workers-kubeadm-config-centos.yaml:7]:
```suggestion
  - nmcli connection load /etc/NetworkManager/system-connections/eth0.nmconnection
  - nmcli connection up eth0
  - nmcli connection load /etc/NetworkManager/system-connections/{{ IRONIC_ENDPOINT_BRIDGE }}.nmconnection
  - nmcli connection up {{ IRONIC_ENDPOINT_BRIDGE }}
```

Comment (Sunnatillo) [reply tests/roles/run_tests/templates/cluster-template-workers-kubeadm-config-centos.yaml:7]: Yes you are right. I somehow missed it. Changed it

File: tests/roles/run_tests/templates/cluster-template-controlplane-kubeadm-config-centos.yaml (hunk 1 above)
Comment (Sunnatillo) [reply tests/roles/run_tests/templates/cluster-template-controlplane-kubeadm-config-centos.yaml]: changed it

File: tests/roles/run_tests/templates/cluster-template-controlplane-kubeadm-config-centos.yaml [hunk 3]
```diff
@@ -1,8 +1,10 @@
 # CentOS specific controlplane kubeadm config
 preKubeadmCommands:
   - systemctl restart NetworkManager.service
-  - nmcli connection load /etc/sysconfig/network-scripts/ifcfg-eth0
-  - nmcli connection up filename /etc/sysconfig/network-scripts/ifcfg-eth0
+  - nmcli con load /etc/NetworkManager/system-connections/eth0.nmconnection
+  - nmcli connection up filename /etc/NetworkManager/system-connections/eth0.nmconnection
```
Comment (lentzi90) [inline tests/roles/run_tests/templates/cluster-template-controlplane-kubeadm-config-centos.yaml:5]: Are you sure we should have `filename` here? I can see that it is the same way we used to have it, but I don't find this in the documentation (`nmcli con up --help`). Perhaps it is for some older version? I think it should be `nmcli con up eth0` now.

Comment (Sunnatillo) [reply tests/roles/run_tests/templates/cluster-template-controlplane-kubeadm-config-centos.yaml]: I agree with you. filename was used in here. I removed it now. Changed it to interface name.
Thank you for noticing

**Comment by Sunnatillo** [comment]
test-ubuntu-integration-main

**Review by lentzi90 — COMMENTED** [review]
Great work on this @Sunnatillo !
/lgtm

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Review by smoshiur1237 — APPROVED** [review]

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**